server/src/routes/course.ts

```diff
@@ -153,3 +170,64 @@ router.post(
   handleInvalidRequestJson,
   controllerDispatcher(addCourse)
 );
+
+/**
+ * @swagger
+ * /v1/courses/{courseId}/instances/{instanceId}/grades/csv:
+ *   post:
+ *     tags: [Course]
+ *     description: Add attainment points for users for specific course instance.
+ *     requestBody:
+ *       content:
+ *         multipart/form-data:
+ *     responses:
+ *       200:
+ *         description: Grading csv uploaded and parsed succesfully.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 success:
+ *                   type: boolean
+ *                   description: Success of the request.
+ *       400:
+ *         description: A validation or parsing error with the csv has occurred.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Failure'
+ *       401:
+ *         description: The requester is not logged in.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Failure'
+ *       403:
+ *         description: >
+ *           The requester is not authorized to add grading data
+ *           to the course instance (not teacher in the course).
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Failure'
+ *       404:
+ *         description: Course, course instance, or any of the attainable does not exist.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Failure'
+ *       409:
+ *         description: >
+ *           Course instance does not belong to the course or
+ *           study attainment does not belong to the course instance.
```

Comment:
There is no study attainment in the route, unless this also occurs when the CSV tags are incorrect.

Reply:
Yes this is incase any of the attainment id's in the CSV file do not belong to the instance.